src/wasm/wasm-binary.cpp

```diff
@@ -3122,12 +3122,15 @@ void WasmBinaryBuilder::readDataSegments() {
     }
     curr->setName(Name::fromInt(i), false);
     curr->isPassive = flags & BinaryConsts::IsPassive;
-    Index memIdx = 0;
-    if (flags & BinaryConsts::HasIndex) {
-      memIdx = getU32LEB();
-    }
-    memoryRefs[memIdx].push_back(&curr->memory);
-    if (!curr->isPassive) {
+    if (curr->isPassive) {
+      curr->memory = Name();
+      curr->offset = nullptr;
+    } else {
+      Index memIdx = 0;
+      if (flags & BinaryConsts::HasIndex) {
+        memIdx = getU32LEB();
+      }
+      memoryRefs[memIdx].push_back(&curr->memory);
```

Comment:
Is this a separate change?

Reply:
No, the test I added depends on being able to have passive segments without any memories. The previous code incorrectly segfaulted because it unconditionally assumed there was a memory available.

Reply:
I see, thanks!